src/apps/toolbar/modules/Notifications/infra/ArrivalPreview.tsx

```diff
@@ -0,0 +1,93 @@
+import { invoke } from '@tauri-apps/api/core';
+import { Button } from 'antd';
+import { AnimatePresence, motion } from 'framer-motion';
+import moment from 'moment';
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { SeelenCommand } from 'seelen-core';
+
+import { BackgroundByLayersV2 } from '../../../../seelenweg/components/BackgroundByLayers/infra';
+
+import { Selectors } from '../../shared/store/app';
+
+import { Icon } from '../../../../shared/components/Icon';
+
+// Difference between Windows epoch (1601) and Unix epoch (1970) in milliseconds
+const EPOCH_DIFF_MILLISECONDS = 11644473600000n;
+
+function WindowsDateFileTimeToDate(fileTime: bigint) {
+  return new Date(Number(fileTime / 10000n - EPOCH_DIFF_MILLISECONDS));
+}
+
+export function ArrivalPreview() {
+  const notifications = useSelector(Selectors.notifications);
+  const [ currentNotificationPreviewSet, setCurrentNotificationPreviewSet ] = useState(false);
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
```

Comment:
there is a utility function on the project called `useTimeout` use that please.

Reply:
I haven't found that... could you please refer which one you would like to use, or from where to import it? 
(autoimport did not found that, i did not found an example refference, or usage)

![image](https://github.com/user-attachments/assets/525ef8e5-ec07-4473-8d0e-41a5d866ef78)

if you can tell me how to find that, that also helps me. It was a bit long ago, when I last touched react...

Reply:
oh sorry it was `useInterval` in this case intead creating a timeout directly use lodash debounce it will cancel the last call if a new one is make it.

Reply:
@luriusTM you only need change this one and the pr will be merged :D

Reply:
Added the new hook, and fixed the line ending conflict :)